tests/nosetests/pyanaconda_tests/modules/storage/module_storage_checker_test.py

```diff
@@ -108,3 +115,117 @@ def lvm_verification_test(self):
+
+        with patch_open(mock_open(read_data="5.10.50-openpower1-p59fd803")):
+            self.assertEqual(_get_opal_firmware_kernel_version(), "5.10.50-openpower1-p59fd803")
+
+        with patch_open(mock_open(read_data="v4.15.9-openpower1-p9e03417")):
+            self.assertEqual(_get_opal_firmware_kernel_version(), "4.15.9-openpower1-p9e03417")
+
+    def check_opal_firmware_kernel_version_test(self):
+        """Test the function for checking the firmware kernel version."""
+        check = partial(_check_opal_firmware_kernel_version)
+
+        self.assertFalse(check(None, None))
+        self.assertFalse(check("", ""))
+        self.assertFalse(check("5.09", "5.10"))
+        self.assertFalse(check("5.9.1", "5.10"))
+        self.assertFalse(check("5.9.50-openpower1-p59fd803", "5.10"))
+        self.assertFalse(check("5.8", "5.10"))
+        self.assertFalse(check("4.0-openpower1-p59fd803", "5.10"))
+
+        self.assertTrue(check("5.10", "5.10"))
+        self.assertTrue(check("5.10.1", "5.10"))
+        self.assertTrue(check("5.10.50-openpower1-p59fd803", "5.10"))
+        self.assertTrue(check("5.11", "5.10"))
+        self.assertTrue(check("6.0-openpower1-p59fd803", "5.10"))
+
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_arch_test(self, mocked_arch):
+        """Check verify_opal_compatibility with a different arch."""
+        reporter = Mock()
+        mocked_arch.get_arch.return_value = "x86_64"
+
+        verify_opal_compatibility(None, {}, None, reporter)
+        reporter.assert_not_called()
+
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_platform_test(self, mocked_arch):
+        """Check verify_opal_compatibility with a different platform."""
+        reporter = Mock()
+        mocked_arch.get_arch.return_value = "ppc64le"
+        mocked_arch.is_powernv.return_value = False
+
+        verify_opal_compatibility(None, {}, None, reporter)
+        reporter.assert_not_called()
+
+    @patch("pyanaconda.modules.storage.checker.utils._get_opal_firmware_kernel_version")
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_version_test(self, mocked_arch, version_getter):
+        """Check verify_opal_compatibility with a higher firmware version."""
+        reporter = Mock()
+        mocked_arch.get_arch.return_value = "ppc64le"
+        mocked_arch.is_powernv.return_value = True
+        version_getter.return_value = "5.10.50-openpower1-p59fd803"
+
+        verify_opal_compatibility(None, {}, None, reporter)
+        reporter.assert_not_called()
+
+    @patch.object(XFS, "mountable", new_callable=PropertyMock)
+    @patch("pyanaconda.modules.storage.checker.utils._get_opal_firmware_kernel_version")
+    @patch("pyanaconda.modules.storage.checker.utils.arch")
+    def opal_verification_test(self, mocked_arch, version_getter, xfs_mountable):
+        """Check verify_opal_compatibility."""
```

Comment:
And this test is actually for a too low version, but we can't learn that here.

Reply:
Fixed.